### ISSUE TITLE
Modals do not pass through accessibility* props

### ISSUE THREAD

**Issue by paularmstrong** [necolas/react-native-web#2314]
**The problem**

When using a `Modal`, I have a requirement to include at least a `aria-labelledBy` prop that references the title of the dialog. However, no accessibility* props appear to be carried over to the `role="dialog"` element that the `<Modal />` produces. More info on dialog role: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role

**How to reproduce**

Simplified test case: https://codesandbox.io/s/react-native-web-modal-does-not-include-a11y-props-h26vs6?file=/src/App.js

Steps to reproduce:
1. Add `<Modal accessibilityLabelledBy="some-id">...</Modal>`
2. Render

**Expected behavior**

`aria-*` props should be added as provided by `accessibility*` props

**Environment (include versions). Did this work in previous versions?**

* React Native for Web (version): ^0.17.0, ^0.18.1
* React (version): 17, 18
* Browser: any

**Comment by necolas** [comment]
LMK if this fix looks right https://github.com/necolas/react-native-web/pull/2315

**Comment by necolas** [comment]
Should be fixed in 0.18.2
